gloo/docs/user_guides/gateway/tcp_proxy/_index.md

```diff
@@ -0,0 +1,194 @@
+
+The gateway will contain the following: 
+```yaml
+apiVersion: gateway.solo.io/v2alpha1
+kind: Gateway
+metadata:
+  name: tcp
+  namespace: gloo-system
+spec:
+  bindAddress: '::'
+  bindPort: 8000
+  tcpGateway:
+    destinations:
+    - name: one
+      destination:
+        single:
+          upstream:
+            name: gloo-system-tcp-echo-1025
+            namespace: gloo-system
+  useProxyProto: false
+```
+```bash
+pbpaste | k apply -n gloo-system -f -
+```
+
+To check that the gateway has been created properly run:
+```bash
+$ kubectl get gateways.gateway.solo.io -A
+
+NAMESPACE     NAME          AGE
+gloo-system   gateway       26h
+gloo-system   gateway-ssl   26h
+gloo-system   tcp           5s
+```
+
+
+The above gateway will be read in by gloo, which will combine it with the other gateways into a Proxy resource.
+To make sure that the configuration has been translated properly run:
+```bash
+$ kubectl get proxies.gloo.solo.io -n gloo-system -oyaml
+
+apiVersion: v1
+items:
+- apiVersion: gloo.solo.io/v1
+  kind: Proxy
+  metadata:
+    creationTimestamp: "2019-07-05T16:01:14Z"
+    generation: 48
+    labels:
+      created_by: gateway
+    name: gateway-proxy
+    namespace: gloo-system
+    resourceVersion: "150947"
+    selfLink: /apis/gloo.solo.io/v1/namespaces/gloo-system/proxies/gateway-proxy
+    uid: 1d7b84d2-9f3e-11e9-8766-3e49e0c5bb1c
+  spec:
+    listeners:
+    - bindAddress: '::'
+      bindPort: 8000
+      name: listener-::-8000
+      tcpListener:
+        tcpHosts:
+        - destination:
+            single:
+              upstream:
+                name: gloo-system-tcp-echo-1025
+                namespace: gloo-system
+          name: one
+      useProxyProto: false
+  status:
+    reported_by: gloo
+    state: 1
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+```
+
+If the translation worked than the listeners array in the resource spec will contain an entry for the tcp service we will be routing to.
+Once the state on the resource is recorded as `1` the service is ready to be routed to.
+
+The next step is adding a port to the gateway-proxy service so we can route to the envoy listener which is handling our TCP traffic.
+
+The service should look like the following:
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: gloo
+    gloo: gateway-proxy
+  name: gateway-proxy
+  namespace: gloo-systemservices/gateway-proxy
+  uid: 1b624541-9f3e-11e9-8766-3e49e0c5bb1c
+spec:
+  externalTrafficPolicy: Cluster
+  ports:
+  - name: http
+    nodePort: 32302
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  - name: https
+    nodePort: 30440
+    port: 443
+    protocol: TCP
+    targetPort: 8443
+  - name: tcp
+    nodePort: 30197
+    port: 8000
+    protocol: TCP
+    targetPort: 8000
+  selector:
+    gloo: gateway-proxy
+  sessionAffinity: None
+  type: LoadBalancer
+status:
+  loadBalancer: {}
+```
```

Comment:
add highlights for the relevant parts:
see here: https://gohugo.io/content-management/syntax-highlighting/